frontend/mr-admin-flate/src/components/tiltaksgjennomforinger/opprett/OpprettTiltaksgjennomforingContainer.tsx

```diff
@@ -0,0 +1,244 @@
+      antallPlasser: data.antallPlasser,
+      tiltakstypeId: data.tiltakstype,
+      enhet: data.enhet,
+      navn: data.tittel,
+      sluttDato: formaterDatoSomYYYYMMDD(data.tilDato),
+      startDato: formaterDatoSomYYYYMMDD(data.fraDato),
+      avtaleId: data.avtale,
+      virksomhetsnummer: avtale?.leverandor?.organisasjonsnummer,
+      tiltaksnummer: `${data.tiltaksnummer}#${data.aar}`
+    };
+
+    try {
+      const response = await mulighetsrommetClient.tiltaksgjennomforinger.opprettTiltaksgjennomforing({
+        requestBody: body,
+      });
+      props.setResult(response.id);
+    } catch {
+      props.setError("Klarte ikke opprette gjennomføring");
+    }
+  };
+
+  const avtalerOptions = () => {
+    if (isLoadingAvtaler || !avtaler) {
+      return <option value={""}>Laster...</option>;
+    }
+    if (avtaler.data.length === 0) {
+      return <option value={""}>Ingen avtaler funnet</option>;
+    }
+    return <>
+      <option value={""}>Velg en</option>
+        { avtaler.data.map((avtale: Avtale) =>
+          <option key={avtale.id} value={avtale.id}>
+            {avtale.navn}
+          </option>
+        )}
+    </>;
+  }
+
+  return (
+    <FormProvider {...form}>
+      <form onSubmit={handleSubmit(postData)}>
+        <FormGroup>
+          <Select
+            label={"Tiltakstype"}
+            error={errors.tiltakstype?.message}
+            {...register("tiltakstype", {
+              onChange: (e) => {
+                setAvtaleFilter({ ...aFilter, tiltakstype: e.target.value })
+                setAvtale(undefined);
+              }})
+            }
+          >
+            {isLoadingTiltakstyper || !tiltakstyper
+              ? <option value={""}>Laster...</option>
+              : <>
+                <option value={""}>Velg en</option>
+                {tiltakstyper.data
+                  .map((tiltakstype) => (
+                    <option key={tiltakstype.id} value={tiltakstype.id}>
+                      {tiltakstype.navn}
+                    </option>
+                  ))
+                }
+              </>
+            }
+          </Select>
+          <Select
+            label={"Avtale"}
+            error={errors.avtale?.message}
+            {...register("avtale", {
+              onChange: (e) => {
+                const selectedAvtale = avtaler?.data.find((avtale: Avtale) => avtale.id === e.target.value);
+                setAvtale(selectedAvtale);
+              }
+            }
+            )}
+          >
+            {avtalerOptions()}
+          </Select>
+        </FormGroup>
+        <FormGroup cols={2}>
+          <TextField
+            error={errors.tittel?.message}
+            label="Tiltaksnavn"
+            {...register("tittel")}
+          />
+          <div></div>
+          <TextField
+            error={errors.tiltaksnummer?.message}
+            label="Tiltaksnummer"
+            {...register("tiltaksnummer", { valueAsNumber: true })}
+          />
```

Comment:
Vi må finne ut hvordan Arena genererer tiltaksnummer i dag. Det bør ikke være noe bruker selv kan skrive inn, tenker jeg.

Reply:
https://nav-it.slack.com/archives/C01DE03F4LS/p1681205863419159